crates/bevy_window/src/window.rs

```diff
@@ -511,6 +518,22 @@ impl Window {
     pub fn is_focused(&self) -> bool {
         self.focused
     }
+
+    #[inline]
+    pub fn icon(&self) -> Option<WindowIcon> {
+        self.icon.clone()
+    }
+
+    pub fn set_icon(&mut self, icon: Option<WindowIcon>) {
```

Comment:
having this take an `Option<WindowIcon>` feels a little weird.
Especially if the name of the function is `set_icon` I would expect this to set the icon to a valid icon.

Imo either
1) have a
```rust
pub fn set_icon(&mut self, icon: WindowIcon);
pub fn remove_icon(&mut self);
```

Or (i think this might be my preferred)
```rust
pub fn set_icon(&mut self, icon: Into<Option<WindowIcon>>);
```
this way you don't have to wrap the argument in a `Some`.

Reply:
I was trying to keep the format of 1 getter +  1 setter, but I agree that it does not work very well in this case. 
I am open to either solution. 
Second one looks more interesting though, so I will try to implement that once the dependency problem was fixed and the example actually works.

Reply:
Yeah I think I prefer making set_icon take a normal handle, just because setting the icon is probably the 99.9% case.